src/main/java/org/zalando/nakadi/security/ClientResolver.java

```diff
@@ -50,15 +49,25 @@ public Client resolveArgument(final MethodParameter parameter,
             return new FullAccessClient(clientId.orElse(FULL_ACCESS_CLIENT_ID));
         }
 
-        return clientId.map(client -> new NakadiClient(client, getScopes()))
+        return clientId.map(client -> new NakadiClient(client, getRealm()))
                 .orElseThrow(() -> new UnauthorizedUserException("Client unauthorized"));
     }
 
-    private Set<String> getScopes() {
+    public String getRealm() {
+        String realmString = "";
+
         final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        if (authentication instanceof OAuth2Authentication) {
-            return ((OAuth2Authentication) authentication).getOAuth2Request().getScope();
+        if (OAuth2Authentication.class.equals(authentication.getClass())) {
+            final Object details = ((OAuth2Authentication) authentication).getUserAuthentication().getDetails();
+            if (details instanceof Map) {
```

Comment:
do you think we could avoid 2 additional inference of type here? it is gonna be done on a lot requests, but does it really necessary?

Reply:
That sounds like a good idea. I'll give it a try.

Reply:
What do you think about a5c3ded

Reply:
What do you think about a5c3ded